### ISSUE TITLE
Using it on OSX

### ISSUE THREAD

**Issue by WarrenFaith** [Moonzel/Godot-PerfBullets#2]
Hey, 

I wanted to try it in a fresh project, so I used the internal AsserLib to get this one but it fails to load.
This is the error output:

> Can't open dynamic library: /Users/.../workspace/git/.../addons/PerfBullets/bin/spawner.macos.template_debug.framework. Error: dlopen(/Applications/Godot.app/Contents/MacOS/../Frameworks/spawner.macos.template_debug.framework, 0x0002): tried: '/Applications/Godot.app/Contents/MacOS/../Frameworks/spawner.macos.template_debug.framework' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Applications/Godot.app/Contents/MacOS/../Frameworks/spawner.macos.template_debug.framework' (no such file), '/Applications/Godot.app/Contents/MacOS/../Frameworks/spawner.macos.template_debug.framework' (no such file).
  core/extension/gdextension.cpp:455 - GDExtension dynamic library not found: /Users/.../workspace/git/.../addons/PerfBullets/bin/spawner.macos.template_debug.framework
  Failed loading resource: res://addons/PerfBullets/bin/spawner.gdextension. Make sure resources have been imported by opening the project in the editor at least once.

I see that there are files missing but I am not sure what to do to fix this. I also read in the readme that this is prebuilt for Windows but I haven't seen a guide to built it for OSX.

Any help appreciated!

**Comment by Moonzel** [comment]
Thanks for leaving the error logs. I have only prebuilt windows in the plugin so far, as that is the only platform I have available to test with my hardware. If you look towards the bottom of the README, I have put directions on how to edit/build the plugin, and the same instructions should apply for OSX. If they do not, please comment here.

**Comment by Moonzel** [comment]
If there are any further problems, please let me know.

**Comment by blabtonic** [comment]
Same error happens on Linux as well

**Comment by Dan-Mizu** [comment]
Yes it seems this addon only seems to work with Windows at the moment. I work on my Godot projects on different platforms and due to this I can't work on projects utilizing this addon on MacOS. 

It is a shame because I LOVE this addon so far as the dev experience has been great, and I lack the skills needed to build this on MacOS- and the knowledge to understand why certain gdextension addons fail to run on different platforms. If anyone else figures it out or creates a successful build I would love to know!

**Comment by Moonzel** [comment]
I am sorry to see that this is still occurring. The way gdextension is set up should allow this to work seamlessly on MacOS and Linux. Has anyone on Mac or Linux tried building the project through command line? Since all I have is a Windows PC, it makes it extremely difficult for me to debug other OSs.

**Comment by PedroG98** [comment]
Hello, I managed to run the plugin on Linux. Note that I'm not sure which of the following steps are necessary:

- Download scons and godot-cpp
- Place the godot-cpp folder inside addons/PerfBullets (as instructed in the instructions to edit the plugin)
- Compile godot-cpp for linux by running scons on addons/PerfBullets/godot-cpp (Don't know if this step was necessary)
- Compile PerfBullets for linux by running scons on addons/PerfBullets
- The file libspawner.linux.template_debug.x86_64.so will be generated in addons/PerfBullets/bin
- **The file addons/PerfBullets/spawner.gdextension needs to be changed, because it only accepts sharedlibs with the name spawner.platform**. Open it and write libspawner instead of spawner for your platform of choice (in this case line 14)
-  Open godot and the library should work

It's a bit confusing, but I hope it helps.

**Comment by Moonzel** [comment]
This is great progress! Thank you for writing this out. I see how this would be an issue, as I was not aware that the name changed per OS. I will attempt to update the spawner.gdextension file with the new name, but I need somebody on MacOS to try these steps to get the name on MacOS. @Dan-Mizu, would you be willing to try?

**Comment by Ughuuu** [comment]
I mean, I can add a PR and add automated actions to this. I recently merged the build and sign actions into godot-cpp-template https://github.com/godotengine/godot-cpp-template?tab=readme-ov-file#usage---actions

**Comment by Moonzel** [comment]
If you think that would fix it, go right ahead! I do not know as much about this aspect of GDExtension.

**Comment by Ughuuu** [comment]
Do you have a apple developer account though? As after those steps(in the action I linked), you will still need to sign it.

**Comment by Moonzel** [comment]
I was not aware I needed that, I do not have it. Is there any other way you can think of to debug this?

**Comment by Ughuuu** [comment]
If you dont sign it it will work on mac(and other platforms) but on mac it will give error and you have to click bypass security.
Ill create a branch for you to review this week.

**Comment by Ughuuu** [comment]
Also what I propose is basically actions that would build for all platforms after every commit, eg. similar to this: https://github.com/appsinacup/godot-rapier-2d/actions

**Comment by Moonzel** [comment]
I will be honest with you, I am not very familiar with Github and its actions. Please make a PR for your ideas, as they seem to make sense, just make sure to update the paths on any of the GDExtension files that need a path (this has caused problems in the past).

**Comment by Ughuuu** [comment]
On it!

**Comment by Moonzel** [comment]
I know it has been quite a long time, life caught up to me, but @Ughuuu, did you ever make any progress? If not, who is still experiencing this problem?

**Comment by Ughuuu** [comment]
Oh no, sorry, also got caught in other things.

**Comment by Moonzel** [comment]
All good, if you wanted to do as you suggested, it would be appreciated. If you cannot, I would need someone to help with this as I am still not familiar with specifics of Linux and MacOS.

**Comment by Ughuuu** [comment]
There is quite some work to support on all platforms I'm afraid. Would you be interested in porting this to a module instead? While it would up the barrier for using for people, it would at least support all platforms and you won't need to build for everything (win, linux, mac, web, ios, android).
However, people would need to build it themselves to use it, so it would definitely be a downside.
You got a discord if u wanna talk more?

**Comment by Moonzel** [comment]
I do have discord (moonzel), I do not know what porting to a module would entail, but I am hesitant to raise the barrier for entry for this project.

**Comment by dimonenka** [comment]
I followed this on my macbook and it worked (to my surprise)! I think this could be pushed to the readme (separate from the "editin plugin" section) with a caveat that this is a general recipe but there might be system-dependent nuances. Eg after compiling with Scons, in the bin folder I had a folder that contained "libspawner.macos.template_debug", and I had to move it up to the bin folder and respectively change the macos.debug line in spawner.gdextension.

> Hello, I managed to run the plugin on Linux. Note that I'm not sure which of the following steps are necessary:
> 
> * Download scons and godot-cpp
> * Place the godot-cpp folder inside addons/PerfBullets (as instructed in the instructions to edit the plugin)
> * Compile godot-cpp for linux by running scons on addons/PerfBullets/godot-cpp (Don't know if this step was necessary)
> * Compile PerfBullets for linux by running scons on addons/PerfBullets
> * The file libspawner.linux.template_debug.x86_64.so will be generated in addons/PerfBullets/bin
> * **The file addons/PerfBullets/spawner.gdextension needs to be changed, because it only accepts sharedlibs with the name spawner.platform**. Open it and write libspawner instead of spawner for your platform of choice (in this case line 14)
> * Open godot and the library should work
> 
> It's a bit confusing, but I hope it helps.

**Comment by nodepond** [comment]
Amazing, this also worked on macOS.

For the sake of completeness:
In **addons/PerfBullets/spawner.gdextension** this line needs to get changed from

`macos.debug = "res://addons/PerfBullets/bin/spawner.macos.template_debug.framework/"`

to

`macos.debug = "res://addons/PerfBullets/bin/spawner.macos.template_debug.framework/libspawner.macos.template_debug"`
